pkg/types/request.go

```diff
@@ -53,7 +58,7 @@ type RetrievalRequest struct {
 // and writing and it is explicitly set to be trusted (i.e. it will not
 // check CIDs match bytes). If the storage is not truested,
 // request.LinkSystem.TrustedStore should be set to false after this call.
-func NewRequestForPath(store ReadableWritableStorage, cid cid.Cid, path string, full bool) (RetrievalRequest, error) {
+func NewRequestForPath(store ReadableWritableStorage, cid cid.Cid, path string, full bool, protocols []multicodec.Code) (RetrievalRequest, error) {
```

Comment:
I think I'd rather this just be left alone for this. The struct is public and this is only meant to be a shortcut utility constructor but you could also just manually craft one since it's all public. I think in preloader I make the initial Request with this and then tinker with the `PreloadLinkSystem` property afterward so as not to pollute this common pathway with guff. I think that would apply in this case too since this seems very specific to Saturn, no?

Reply:
somewhat. I mean I actually wrote this cause i was tinkering with a Lassie-k6 plugin so there are other uses. BUT yes I'm fine with dropping it from this utility constructor.

Reply:
fixed!